### PR TITLE
feat(ldap): hot reloading ldap credentials on change

### LINKED ISSUE #2161: [Feat]: zot should watch for changes in the LDAP bind credentials configuration file
### Is your feature request related to a problem? Please describe.

If the user changes the content of the LDAP configuration file added as part of https://github.com/project-zot/zot/pull/1778 zot needs to be restarted to pick the configuration change

### Describe the solution you'd like

If zot is notified the ldap credentials file has changed on the disk, zot starts using the new configuration right away. Same as for zot config file changes.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:
Closes #2161

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cli/server/config_reloader.go [hunk 1]
```diff
@@ -73,13 +76,26 @@ func (hr *HotReloader) Start() {
...
-						err := LoadConfiguration(newConfig, hr.filePath)
+						err := LoadConfiguration(newConfig, hr.configPath)
 						if err != nil {
 							log.Error().Err(err).Msg("failed to reload config, retry writing it.")
 
 							continue
 						}
 
+						if hr.ctlr.Config.HTTP.Auth != nil && hr.ctlr.Config.HTTP.Auth.LDAP != nil {
+							err = hr.watcher.Remove(hr.ctlr.Config.HTTP.Auth.LDAP.CredentialsFile)
```
Comment (peusebiu) [inline pkg/cli/server/config_reloader.go:87]: why always remove credential files?
this is needed only if it changed in the meantime, (hr.ctlr.Config.HTTP.Auth.LDAP.CredentialsFile != newConfig.HTTP.Auth.LDAP.CredentialsFile) right?

Comment (laurentiuNiculae) [reply pkg/cli/server/config_reloader.go:87]: I think that's right

**Review by andaaron — DISMISSED** [review]
LGTM

File: pkg/api/config/config.go [hunk 2]
```diff
@@ -19,7 +19,6 @@ var (
 
 	openIDSupportedProviders = [...]string{"google", "gitlab", "oidc"} //nolint: gochecknoglobals
 	oauth2SupportedProviders = [...]string{"github"}                   //nolint: gochecknoglobals
-
```
Comment (rchincha) [inline pkg/api/config/config.go:22]: undo this?

**Review by andaaron — APPROVED** [review]

**Pull request merged**